GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -2503,34 +2503,34 @@ private static Image GetItemImage(SubmoduleInfo info)
             return info.IsDirty ? Resources.IconSubmoduleDirty : Resources.Modified;
         }
 
-        private static void GetSubmoduleStatusAsync(SubmoduleInfo info, CancellationToken cancelToken)
+        private static async Task GetSubmoduleStatusAsync(SubmoduleInfo info, CancellationToken cancelToken)
         {
-            Task.Factory.StartNew(() =>
-            {
-                var submodule = new GitModule(info.Path);
-                var supermodule = submodule.SuperprojectModule;
-                var submoduleName = submodule.GetCurrentSubmoduleLocalPath();
+            await TaskScheduler.Default;
+            cancelToken.ThrowIfCancellationRequested();
 
-                info.Status = null;
+            var submodule = new GitModule(info.Path);
+            var supermodule = submodule.SuperprojectModule;
+            var submoduleName = submodule.GetCurrentSubmoduleLocalPath();
 
-                if (string.IsNullOrEmpty(submoduleName) || supermodule == null)
-                {
-                    return;
-                }
+            info.Status = null;
 
-                var submoduleStatus = GitCommandHelpers.GetCurrentSubmoduleChanges(supermodule, submoduleName);
-                if (submoduleStatus != null && submoduleStatus.Commit != submoduleStatus.OldCommit)
-                {
-                    submoduleStatus.CheckSubmoduleStatus(submoduleStatus.GetSubmodule(supermodule));
-                }
+            if (string.IsNullOrEmpty(submoduleName) || supermodule == null)
+            {
+                return;
+            }
 
-                if (submoduleStatus != null)
-                {
-                    info.Status = submoduleStatus.Status;
-                    info.IsDirty = submoduleStatus.IsDirty;
-                    info.Text += submoduleStatus.AddedAndRemovedString();
-                }
-            }, cancelToken, TaskCreationOptions.AttachedToParent, TaskScheduler.Default);
+            var submoduleStatus = GitCommandHelpers.GetCurrentSubmoduleChanges(supermodule, submoduleName);
+            if (submoduleStatus != null && submoduleStatus.Commit != submoduleStatus.OldCommit)
```

Comment:
`if (submoduleStatus == null) { return; }` will simplify the following code and reduce nesting

Reply:
This is the original implementation - compare ignoring whitespaces.